part2/stages/Functions/install-main

```diff
@@ -872,16 +872,27 @@ install_file()
     local SRC="$1"
     local DST="$2"
     local PACKAGE_TYPE="$3"
+    local config_mounted="0"
+    local RET="0"
 
-    mkdir -p $(dirname "${DOM0_MOUNT}/${DST}") || return 1
+    if [ "$( expr substr "$DST" 1 8 )" = "/config/" ] ; then
+        config_mounted="1"
```

Comment:
I would set this after you call ```mount_config```. Additionally I would also make the ```mount_config``` call with ``` || RET=1```  like on line 418 (similar), and then check to see it was successful before proceeding.

Reply:
Good call on the config_mounted, but I guess it gets dropped if is_mounted is used.  I'll make it `mount_config || return 1` since no clean up is needed.